### PR TITLE
feat: Upgrade base project for flutter v3

### PR DESCRIPTION
**Pull request by KotonoSora**
- Upgrade all dependencies to latest version
- Upgrade base android with gradle latest version
- Resolve some bug null safety
- Replace dependency flutter_tags to flutter_tags_x
- Resolve bug load widget when using dependency popup_menu

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -1,5 +1,3 @@
 org.gradle.jvmargs=-Xmx1536M
-android.enableR8=true
 android.useAndroidX=true
 android.enableJetifier=true
-android.bundle.enableUncompressedNativeLibs=false
```
Comment (khoadng) [inline android/gradle.properties:5]: This option is needed for targeting Android 6.0, don't remove it.

https://docs.flutter.dev/deployment/android#building-the-app-for-release

Comment (KotonoSora) [reply android/gradle.properties:5]: I reverted it as you requested

**Comment by KotonoSora** [comment]
Revert config as request by owner reviewer
- Revert config for fix app crashes on certain devices on Android 6.0
- Revert config pubspec dependencies don't use the static version

File: lib/boorus/danbooru/presentation/features/accounts/login/widgets/login_box_widget.dart [hunk 2]
```diff
@@ -213,13 +213,13 @@ class LoginField extends HookWidget {
...
   @override
   Widget build(BuildContext context) {
-    final _controller = useTextEditingController();
+    final controller = useTextEditingController();
 
     return TextFormField(
       onChanged: onChanged,
       obscureText: obscureText,
       validator: validator,
-      controller: controller ?? _controller,
```
Comment (khoadng) [inline lib/boorus/danbooru/presentation/features/accounts/login/widgets/login_box_widget.dart:222]: if you remove the null check here and rename the controller, please remove this too and make the controller non-nullable.
https://github.com/khoadng/Boorusama/blob/ff79645091738c086e43daf736714f2aec137067/lib/boorus/danbooru/presentation/features/accounts/login/widgets/login_box_widget.dart#L216

**Comment by khoadng** [comment]
LGTM

**Pull request merged**